src/pages/commits/ActiveCommits/utils.ts

```diff
@@ -154,26 +154,22 @@ export const getStatusesWithZeroCount = (colors: ColorCount[]) => {
   );
 };
 
-export const hoverTaskIcons = () => {
-  const property = "data-task-icon";
-
-  // find all icons on page
-  const icons = document.querySelectorAll<HTMLElement>(`div[${property}]`);
-  // define mouseover and mouseout behavior for all icons
-  for (let i = 0; i < icons.length; i++) {
-    icons[i].onmouseover = () => {
-      for (let j = 0; j < icons.length; j++) {
-        if (
-          icons[j].getAttribute(property) !== icons[i].getAttribute(property)
-        ) {
-          icons[j].style.opacity = "0.25";
-        }
-      }
-    };
-    icons[i].onmouseout = () => {
-      for (let k = 0; k < icons.length; k++) {
-        icons[k].style.opacity = "1";
-      }
-    };
-  }
+// Functions for injecting and removing style for hovering on task icons
+const taskIconStyle = "task-icon-style";
+
+export const removeGlobalStyle = () => {
+  document.getElementById(taskIconStyle)?.remove();
+};
+
+export const injectGlobalStyle = (taskIdentifier: string) => {
+  removeGlobalStyle();
+
+  const hoverStyle = document.createElement("style");
+  hoverStyle.id = taskIconStyle;
+  hoverStyle.innerHTML = `
+    div[data-task-icon]:not([data-task-icon="${taskIdentifier}"]) {
```

Comment:
What do you think about this? Normally css selector performance is really good but it might make a difference on a page like this with thousands of dom nodes. https://csswizardry.com/2011/09/writing-efficient-css-selectors/

Reply:
Initially I was using the `classNames` instead of an attribute, but it gets attached to some other fluff due to the `styled` library, like this:

<img width="614" alt="Screen Shot 2022-04-25 at 1 43 50 PM" src="https://user-images.githubusercontent.com/47064971/165144323-f2874483-6227-4ff6-8800-f639197c9c1c.png">

Arjun also weighed in on it on this comment [here](https://github.com/evergreen-ci/spruce/pull/1179#discussion_r838557151).

---

I could add an `ID` to the surrounding container  and do something like `#commits-build-variants-container div[data-task-icon]:not([data-task-icon="${taskIdentifier}"])` but I'm not sure how much it would help considering that the attribute portion is supposed to be the slow part.

But also, since this article was written 11 years ago when browsers weren't as optimized as they were today, I wonder if it's something even worth worrying about?